### PR TITLE
Improvement of links

### PR DESCRIPTION
**Pull request by phoenixr-codes**
This pull request improves some links in the book:

- link to exact section if referred to one
- link to new repository of mdbook (old one redirects to the new one)

### CODE REVIEW

File: book/src/overview.md [hunk 1]
```diff
@@ -74,13 +74,13 @@ This block will error
...
 
-You can also configure the build to fail loudly, by setting `on_failure = "bail"` in `book.toml`. See the [configuration reference](./reference.md) for more details.
+You can also configure the build to fail loudly, by setting `on_failure = "bail"` in `book.toml`. See the [configuration reference](./reference.md#booktoml-configuration) for more details.
 
 ### Additional Options
 
 You can pass additional options to each block. The options are structured as TOML key-value pairs.
 
-Note that some options can be passed globally, through the `default` section in `book.toml`. See the [configuration reference](./reference.md) for more details.
+Note that some options can be passed globally, through the `default` section in `book.toml`. See the [configuration reference](./reference.mdbooktoml-configuration) for more details.
```
Comment (tommilligan) [inline book/src/overview.md]: I think you're missing a `#` here
```suggestion
Note that some options can be passed globally, through the `default` section in `book.toml`. See the [configuration reference](./reference.mdbook#toml-configuration) for more details.
```

**Review by tommilligan — COMMENTED** [review]
Thanks, agreed this is much better. I think there's just one typo, after that looks good

File: book/src/overview.md (hunk 1 above)
Comment (phoenixr-codes) [reply book/src/overview.md]: Oh yes somehow missed that. But I think the `#` belongs after the `.md` file extension.

```diff
- ./reference.mdbook#toml-configuration
+ ./reference.md#booktoml-configuration
```

Links to headers do not include backticks (`` ` ``) or periods (`.`) which is the reason why it looks that weird.

**Review by tommilligan — APPROVED** [review]

File: book/src/overview.md (hunk 1 above)
Comment (tommilligan) [reply book/src/overview.md]: Ah of course, thanks

**Pull request merged**